src/Soil-Serializer/SoilObjectTable.class.st

```diff
@@ -0,0 +1,32 @@
+Class {
+	#name : #SoilObjectTable,
+	#superclass : #Object,
+	#instVars : [
+		'lastIndex',
+		'index'
+	],
+	#category : #'Soil-Serializer'
+}
+
+{ #category : #api }
+SoilObjectTable >> add: anObject [
+	index
+		at: anObject
+		ifAbsentPut: [ lastIndex := lastIndex + 1 ]
+]
+
+{ #category : #api }
+SoilObjectTable >> identityIndexOf: anObject [
+	 ^ index at: anObject ifAbsent: [ 0 ]
+]
+
+{ #category : #initialization }
+SoilObjectTable >> initialize [
+	lastIndex := 0.
+	index := IdentityDictionary new
+]
+
+{ #category : #accessing }
+SoilObjectTable >> size [
+	^ lastIndex
```

Comment:
Is this much faster than "index size"?

Reply:
ah, good idea...the size here is just used in tests, but using the size of of the dictioary instead of managing lastIndex would be nice. (in the add: method)